### ISSUE TITLE
Research Scoped API Keys with Canvas

### ISSUE THREAD

**Issue by pushyamig** [tl-its-umich-edu/canvas-course-manager-next#2]
Since we are thinking of using JS Tech stack we might not have API directory wrapper written in JS as in Python. Using the API directory have a benefit of using the Service account of Canvas but not necessarily as great choice. So need to find out if CCM API needs are sufficient for Scoped API

**Comment by pushyamig** [comment]
Research Analysis:
The scoped token scopes the token based on the API selected during the API key configuration and user privileges in canvas.
1. Created an API KEY scoping to “url:POST|/api/v1/accounts/:account_id/users”
2. Created a Token as me and try to add user to canvas ( as that is the  only API I scoped the Dev key). Result: I was able to add user to Canvas
3. Used the same token and tried on different API ( like getting course sections) that is not part of the scope. I got 401 Unauthorized "message": "Insufficient scopes on access token." . Which is good
4. Created a Token as test instructor ( Instx) and tried to add user to Canvas and I received 403 Forbidden error. Which is good way to tighten the access to canvas based on the role.
